mountpoint-s3-fs/src/data_cache/disk_data_cache.rs

```diff
@@ -497,37 +500,51 @@ impl DataCache for DiskDataCache {
         if block_offset != block_idx * self.config.block_size {
             return Err(DataCacheError::InvalidBlockOffset);
         }
-
+        let start = Instant::now();
         let bytes_len = bytes.len();
         let block_key = DiskBlockKey::new(&cache_key, block_idx);
         let path = self.get_path_for_block_key(&block_key);
         trace!(?cache_key, ?path, "new block will be created in disk cache");
 
-        let block = DiskBlock::new(cache_key, block_idx, block_offset, bytes).map_err(|err| match err {
-            DiskBlockCreationError::IntegrityError(_e) => DataCacheError::InvalidBlockContent,
-        })?;
-
-        {
-            let eviction_start = Instant::now();
-            let result = self.evict_if_needed();
-            metrics::histogram!("disk_data_cache.eviction_duration_us")
-                .record(eviction_start.elapsed().as_micros() as f64);
-            result
-        }?;
-
-        let write_start = Instant::now();
-        let (temp_file, size) = self.write_block(&path, block)?;
-        metrics::histogram!("disk_data_cache.write_duration_us").record(write_start.elapsed().as_micros() as f64);
-        metrics::counter!("disk_data_cache.total_bytes", "type" => "write").increment(bytes_len as u64);
-        if let Some(usage) = &self.usage {
-            let mut usage = usage.lock().unwrap();
-            _ = temp_file.persist(path).map_err(|e| e.error)?;
-            usage.add(block_key, size);
-        } else {
-            _ = temp_file.persist(path).map_err(|e| e.error)?;
-        }
+        let put_result = (|| -> DataCacheResult<()> {
```

Comment:
I don't think it's obvious why we've put this into closure and then invoke it. Can you add a comment?